src/app/node-data/component.ts

```diff
@@ -286,6 +311,32 @@ export class NodeDataComponent extends BaseFormValidator implements OnInit, OnDe
     this._cdr.detectChanges();
   }
 
+  private _loadOperatingSystemProfiles() {
+    if (this.isOperatingSystemManagerEnabled) {
+      this.isLoadingOSProfiles = true;
+      const profiles$ = this.isDialogView()
+        ? this._projectService.selectedProject.pipe(take(1)).pipe(
+            switchMap(project => {
+              return this._osmService.getOperatingSystemProfilesForCluster(
+                this._clusterSpecService.cluster.id,
+                project.id
+              );
+            })
+          )
+        : this._datacenterSpec?.spec
+        ? this._osmService.getOperatingSystemProfilesForSeed(this._datacenterSpec.spec.seed)
+        : EMPTY;
+
+      profiles$
+        .pipe(takeUntil(this._unsubscribe))
```

Comment:
Added this so that if API call is pending and component is destroyed, API call is cancelled.